code-for-canada/project-thundercat
# [Quit Test Flow] Completed the Quit Test Flow

# Description

I completed the quit test flow by adding the generic popup box component when selecting the **Quit test** button. This popup box also contains an error/danger message, so I also used the generic system message component to get that.

## Type of change
- New feature (non-breaking change which adds functionality)
- Code cleanliness or refactor

## Screenshot

**English Version:**

![image](https://user-images.githubusercontent.com/23021242/54368661-ebda7e80-464a-11e9-8ec9-9ac168c42d45.png)

**French Version:**

![image](https://user-images.githubusercontent.com/23021242/54368708-044a9900-464b-11e9-9b79-ab488a376e5f.png)

**When less than all the checkboxes are selected (the button still disabled):**

![image](https://user-images.githubusercontent.com/23021242/54368745-13314b80-464b-11e9-880b-7b4956765b7e.png)

**When all checkboxes are selected (the button becomes enabled):**

![image](https://user-images.githubusercontent.com/23021242/54368763-1c221d00-464b-11e9-8352-a9c02f1590bd.png)

***Note that the translation hasn't been done in this PR, except for the buttons. This translation will be done in another PR.**

# Testing

Manual steps to reproduce this functionality:

1.  Start the eMIB sample test.
2.  Complete the pre-test process.
3.  Click **Quit test** and make sure that all buttons / checkboxes in the popup box work well.
4.  Also, make sure that the checkbox states are resetting when closing the popup box (check at least one checkbox before closing the popup box to test that)

Screenshot of unit tests run passing:

![image](https://user-images.githubusercontent.com/23021242/54287308-c894ce00-457b-11e9-8f21-c109cc27c8ad.png)

# Checklist

Applicable for all code changes.
